### DISCUSSION TITLE
QTable show all rows programmatically

### DISCUSSION THREAD

**Discussion by Rolanddoda** [quasarframework/quasar#11592] [General - Components / Directives / etc] [answered]
In [this Codepen](https://codepen.io/Roland1993/pen/dyVMBPg?editors=1010) you can see that I have a simple QTable.

I am showing 3 rows by default, and if the user clicks "Show all rows" then I show all the rows. In the Quasar docs it says that setting `rowsPerPage` to 0 means infinity and thus showing all rows.

I tried setting the `rowsPerPage` to 0 initially and indeed it shows all the rows. But if I change it programmatically, nothing happens plus an infinite loop somehow happens and I can't even refresh the page.

I thought that maybe it's because I have to use `v-mode:pagination` but that didn't help.
Then I tried creating a new `ref` and making `rowsPerPage` inside `pagination` to use a ref. That didn't work either.
I made the `pagination` object a reactive one by using `reactive`, but again, that didn't work.

Is there any way to show all rows programmatically ?

**Comment by Rolanddoda** [comment]
It's worth mentioning, that if I use `v-model:pagination` and set the `pagination.rowsPerPage` to 5 it works. But with 0 it does NOT and somehow the codepen freezes.

**Comment by MilosPaunovic** [comment, marked as answer]
I've just tried to change `<q-btn @click="pagination.rowsPerPage = 5">Show all rows</q-btn>` to equals `0`, and it works properly. Are you certain that it's not an issue of your browser?

**Reply by Rolanddoda** [reply]
Hmmm that's strange! Now it works fine without freezing the browser!

Maybe the issue just wanted me to post a question here and then fix itself ? 🤔 
Pfff

**Reply by MilosPaunovic** [reply]
Anything is possible! Anyways, it's great that your problem is solved 🙂

**Reply by Rolanddoda** [reply]
Yeah, thanks for taking the time looking into the issue

**Reply by MilosPaunovic** [reply]
You're welcome!
